File: src/test/testdata/inspections/unsafeIterators/Queue.java
package main;

import com.badlogic.gdx.utils.Queue;

import java.util.Iterator;

class QueueTest {

  Queue<String> q = new Queue<String>();

  void f(Queue s) {
    <warning>s.iterator()</warning>;
  }

  void g(Iterator i) {
    i.next();
  }

  void test() {

    for (String s: <warning>q</warning>) {

    }

    f(q);
    g(<warning>q.iterator()</warning>);
  }

}

File: src/test/testdata/inspections/unsafeIterators/Queue.kt
package main

import com.badlogic.gdx.utils.Queue

@Suppress("UNUSED_VARIABLE", "UNCHECKED_CAST", "USELESS_CAST")
class QueueTest {

  var q = Queue<String>()

  fun f(s: Queue<String>) {
    <warning>s.iterator()</warning>
  }

  fun g(i: Iterator<Any>) {
    i.next()
  }

  fun test() {

    for (s in <warning>q</warning>) {

    }

    f(q)
    g(<warning>q.iterator()</warning>)
  }

}

